tests/Test/src/test_z.birch

```diff
@@ -1,30 +1,32 @@
 /*
  * Test the normalizing constant of a pdf.
  *
- * - π: The target distribution. 
- * - N: Number of importance samples.
- * - lazy: Use lazy expressions?
+ * @param π The target distribution. 
+ * @param N Number of importance samples.
+ * @param lazy Use lazy expressions?
  */
 function test_z(π:Distribution<Real>, N:Integer, lazy:Boolean) {  
-  /* compute the shape for a t-distribution proposal using the iid samples */
+  /*
+  * compute the shape for a t-distribution proposal using the mean
+  * and sample variance of iid samples, calculated with Welford's online algorithm
+  */
   let k <- 3;  // degrees of freedom
-  let μ <- 0.0;
-  let σ2 <- 0.0;
+  let μ <- 0.0; // running mean
+  let S <- 0.0; // running sum of squares of differences from mean
   for n in 1..N {
     let x <- π.simulate();
-    μ <- μ + x;
-    σ2 <- σ2 + x*x;
+    let dμ <- x - μ;
+    μ <- μ + dμ/n;
+    S <- S + dμ*(x - μ);
   }
-  μ <- μ/N;              // sample mean and location for Student's t
-  σ2 <- σ2/N - μ*μ;      // sample variance
-  σ2 <- σ2/(k/(k - 2));  // spread for Student's t proposal
+  let σ2 <- (S/N)/k; // spread for Student's t proposal
 
   /* draw importance samples */
   z:Real[N];
   parallel for n in 1..N {
     let x <- simulate_student_t(k, μ, σ2);
     let q <- logpdf_student_t(x, k, μ, σ2);
-    let p <- 0.0;
+    p:Real;
```

Comment:
Prefer to initialize, even though unnecessary here, less error prone when code gets modified (esp. copied and pasted!) in future.

Reply:
I see, when reading it I assumed the initialization was unintentional. I'll revert it.

Reply:
```suggestion
    let p <- 0.0;
```